packages/dts-generator/lib/phases/json-fixer.js

```diff
@@ -42,12 +43,38 @@ function addImplicitNamespaces(json) {
 const badData = {
   "sap.ui.test.Opa5": true
 };
+
 function removeBadData(json) {
   json.symbols = _.reject(json.symbols, sym => {
     return badData[sym.name];
   });
 }
 
+/**
+ * In case a interface is tagged as restricted, we should remove it from the classes that uses it
+ * @param json
+ */
+function removeRestrictedInterface(json) {
+  const restrictedInterfaces = [];
+  json.symbols.forEach(symbol => {
+    if (symbol.kind === "interface" && symbol.visibility === "restricted") {
+      restrictedInterfaces.push(symbol.name);
+    }
+  });
+  json.symbols.forEach(symbol => {
```

Comment:
I prefer consistent usage of utility library (_) for high order utils.
So lets use _.forEach